### PR TITLE
feat: prettiers the output to be more inline with clang-tidy

### PR DESCRIPTION
**Pull request by ArchieAtkinson**
Hey,

Stumbled across this repo and really liked how fast it is compared to clang-tidy but wanted the prettier output so I thought I would add it myself. 

Not sure if you're accepting pull requests but I thought I would give you the option. 

This PR prettier ups the output so it's similar to clang-tidy's but not exactly the same. 

Let me know if you have any suggestions or changes you want, I'm not a python person so I might have made some errors. 

Example output below (colour not visible):
```
100%|██████████| 3/3 [00:01<00:00,  2.53it/s]
/home/user/west-workspace/application/src/button_error.cpp:24:7: Error: Do not use 'else' after 'return' [readability-else-after-return]
   22 |      if (errorName.empty()) {
   23 |          return "(unrecognized error)";
   24 |      } else {
      |        ^---
   25 |          return std::string(errorName);
   26 |      }

/home/user/west-workspace/application/tests/button/src/main.cpp:24:5: Error: Misleading indentation: statement is indented too deeply [readability-misleading-indentation]
   22 |      if (x > 0)
   23 |      x++;
   24 |      y++;
      |      ^
   25 |  }
   26 |  
/home/user/west-workspace/application/tests/button/src/main.cpp:22:6: Note: did you mean this line to be inside this 'if' 
   22 |      if (x > 0)
      |       ^

/home/user/west-workspace/application/tests/button/src/main.cpp:18:7: Error: Do not use 'else' after 'return' [readability-else-after-return]
   16 |      if (x > 0) {
   17 |          return x;
   18 |      } else {
      |        ^---
   19 |          return -x;
   20 |      }
   ```

### CODE REVIEW

**Comment by lljbash** [comment]
Hi Archie, sorry for the late reply. I really appriciate your help in improving this project!

I will need some time to test it. For now I only have one doubt: Would those control sequence ruin the output when logs are not printed to a terminal? Should we add a check like `is_tty()` or something?

Thank you again for your effort and feel free to fire any issues or PRs.

File: clangd-tidy [hunk 1]
```diff
@@ -234,6 +311,13 @@ if __name__ == "__main__":
     parser.add_argument(
         "-v", "--verbose", action="store_true", help="Print verbose output from clangd."
     )
+    parser.add_argument(
+        "-c",
+        "--context",
+        type=int,
+        default=2,
+        help="Number of lines of code shown either side of the detected issue. [default: 2]",
```
Comment (lljbash) [inline clangd-tidy:319]: I don't quite understand this help message.

**Comment by ArchieAtkinson** [comment]
> Hi Archie, sorry for the late reply. I really appriciate your help in improving this project!
> 
> I will need some time to test it. For now I only have one doubt: Would those control sequences ruin the output when logs are not printed to a terminal? Should we add a check like `is_tty()` or something?
> 
> Thank you again for your effort and feel free to fire any issues or PRs.

No worries! And yes that's a good point. I'll look into that now

File: clangd-tidy (hunk 1 above)
Comment (ArchieAtkinson) [reply clangd-tidy:319]: This is provides how much code is shown around an error. For example when its the default 2 you might get:
```bash
/home/user/west-workspace/notus/subsys/file_system/file_system.cpp:22:26: Error: Variable name 'a' is too short, expected at least 3 characters [readability-identifier-length]
   20 |      arch_nop();
   21 |  
   22 |      [[maybe_unused]] int a = 0;
      |                           ^
   23 |  }
ninja: build stopped: subcommand failed.
```

when it's 5 you could get:
```bash
/home/user/west-workspace/notus/subsys/file_system/file_system.cpp:22:26: Error: Variable name 'a' is too short, expected at least 3 characters [readability-identifier-length]
   17 |  
   18 |  FileSystem::~FileSystem()
   19 |  {
   20 |      arch_nop();
   21 |  
   22 |      [[maybe_unused]] int a = 0;
      |                           ^
   23 |  }
   24 |  
   25 |  void other_fun()
   26 |  {
   27 |
```

Happy to update the help message if you have any ideas

**Comment by ArchieAtkinson** [comment]
@lljbash pushed that, let me know what you think .

**Comment by lljbash** [comment]
> @lljbash pushed that, let me know what you think .

Looks promising! I would like some more tweaks on my own. I am going to patch your changes and make a new PR. I will get you informed when it's ready.

**Comment by lljbash** [comment]
@ArchieAtkinson #2 is ready based on your changes. Come and see if you like it!

**Pull request closed without merging**